tests/ci/build_run_benchmarks.sh

```diff
@@ -0,0 +1,128 @@
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py aws-lc-prod_fips_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py ossl_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py bssl_bm.json
+
+# check for regressions!
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py aws-lc-prod_bm.csv aws-lc-pr_bm.csv prod_vs_pr.csv
+prod_vs_pr_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py aws-lc-prod_fips_bm.csv aws-lc-pr_fips_bm.csv prod_vs_pr_fips.csv
+prod_vs_pr_fips_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py ossl_bm.csv aws-lc-pr_bm.csv ossl_vs_pr.csv
+ossl_vs_pr_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py bssl_bm.csv aws-lc-pr_bm.csv bssl_vs_pr.csv
+bssl_vs_pr_code="$?"
+
+# upload results to s3
+aws s3 cp aws-lc-pr_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-aws-lc-pr_bm.csv"
+aws s3 cp aws-lc-pr_fips_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-aws-lc-pr_fips_bm.csv"
+aws s3 cp aws-lc-prod_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-aws-lc-prod_bm.csv"
+aws s3 cp aws-lc-prod_fips_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-aws-lc-prod_fips_bm.csv"
+aws s3 cp ossl_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-ossl_bm.csv"
+aws s3 cp bssl_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-bssl_bm.csv"
+
+# upload results to lastest folders in s3
+aws s3 mv aws-lc-pr_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-aws-lc-pr_bm.csv"
+aws s3 mv aws-lc-pr_fips_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-aws-lc-pr_fips_bm.csv"
+aws s3 mv aws-lc-prod_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-aws-lc-prod_bm.csv"
+aws s3 mv aws-lc-prod_fips_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-aws-lc-prod_fips_bm.csv"
+aws s3 mv ossl_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-ossl_bm.csv"
+aws s3 mv bssl_bm.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-prod-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-bssl_bm.csv"
+
+# if any of the results gave an exit code of 5, there's a performance regression
+exit_fail=false
+if [ "${prod_vs_pr_code}" != 0 ]; then
+  aws s3 cp prod_vs_pr.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-prod_vs_pr.csv"
+  aws s3 mv prod_vs_pr.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-bm-framework-pr-bucket/latest/${CPU_TYPE}${NOHW_TYPE}-prod_vs_pr.csv"
```

Comment:
That is a problem - I will add a prefix containing the PR-number to the `latest` folder, so there will be separate `pr/#-latest` folders for each PR in order to prevent this problem.

Reply:
* Each PR can have multiple commits. Is the overwrite expected?
* Is the change pushed?

Reply:
The change has been pushed, here is the current iteration of the code:
```
exit_fail=false
if [ "${prod_vs_pr_code}" != 0 ]; then
  aws s3 cp prod_vs_pr.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-ci-bm-framework-pr-bucket/${COMMIT_ID}/${CPU_TYPE}${NOHW_TYPE}-prod_vs_pr.csv"
  aws s3 mv prod_vs_pr.csv s3://"${AWS_ACCOUNT_ID}-aws-lc-ci-bm-framework-pr-bucket/latest-${PR_NUM}/${CPU_TYPE}${NOHW_TYPE}-prod_vs_pr.csv"
  exit_fail=true
```
As you can see, we upload the regression details into two places. Firstly, we upload the file to a commit folder, which is unique so no overrides will take place. This is for when in the future people are interested in the regression data of this commit in the PR. The second place we upload the file is to a latest/pr/# folder. This is simply to make it easier for whoever is interested to look at the latest regression information for each PR. An override is expected here since we retain historical information elsewhere.

Reply:
> An override is expected here since we retain historical information elsewhere.

Some overrides may not be expected. E.g. when `commit1` is pushed before pushing `commit2`, but commit2 finishes the benchmark job earlier, then the `latest` data is result of benchmark `commit1`. This case can happen because some API latency range is very large, especially the ec2 instance launch API, sometimes the script has to wait for several minutes to let the status become ready.

Reply:
Oh I hadn't considered that. Do you have any recommendations on how to avoid this issue?

Currently, I am thinking that I will have each CodeBuild started by the CI create a metadata file containing the unix timestamp of the system. Before starting the benchmarks, it will check the appropriate S3 folder to see if the metadata file exists (if it doesn't then it uploads its own metadata file and continues normally). If the metadata file exists and contains a smaller unix timestamp than the CodeBuild instance, then the CodeBuild will override the existing file on the S3 with its own and continue normally. However, if the metadata file contains a larger unix timestamp than the CodeBuild, the CodeBuild will proceed with the benchmark but will not upload to the `latest` folders.

^ the above technically won't solve the issue if somehow two codebuilds are launched on two different commits at the exact same time, but I think it does the job. What do you think?

Reply:
Also, as discussed in Slack, the latest folder is not a hard requirement 👍